rust/migrate-wicked/src/tuntap.rs

```diff
@@ -0,0 +1,39 @@
+use agama_server::network::model;
+use serde::{Deserialize, Serialize};
+use serde_with::skip_serializing_none;
+
+#[skip_serializing_none]
+#[derive(Debug, PartialEq, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Tun {
+    pub owner: Option<u32>,
+    pub group: Option<u32>,
+}
+
+#[skip_serializing_none]
+#[derive(Debug, PartialEq, Default, Serialize, Deserialize)]
+#[serde(default)]
+pub struct Tap {
+    pub owner: Option<u32>,
+    pub group: Option<u32>,
+}
+
+impl From<&Tun> for model::ConnectionConfig {
+    fn from(tuntap: &Tun) -> model::ConnectionConfig {
+        model::ConnectionConfig::Tun(model::TunConfig {
+            mode: model::TunMode::Tun,
+            group: tuntap.group.map(|v| v.to_string()),
+            owner: tuntap.owner.map(|v| v.to_string()),
+        })
+    }
+}
+
+impl From<&Tap> for model::ConnectionConfig {
+    fn from(tuntap: &Tap) -> model::ConnectionConfig {
+        model::ConnectionConfig::Tun(model::TunConfig {
+            mode: model::TunMode::Tap,
+            group: tuntap.group.map(|v| v.to_string()),
+            owner: tuntap.owner.map(|v| v.to_string()),
```

Comment:
Just wondering, does this work? I can see in the NM docs that it requires e.g. for owner `The user ID which will own the device`. Is `ID` in this case really a u32 in string form or the actual name of the group and owner?

Reply:
I would say  yes, see: https://gitlab.freedesktop.org/NetworkManager/NetworkManager/blob/main/src/libnm-core-impl/nm-setting-tun.c#L169

Reply:
You are right, seems like it is. Weird that it is implemented as a string in dbus.